### PR TITLE
FALCON-1980 : Change input and output argument order for Spark process workflow

### PR DESCRIPTION
**Pull request by peeyushb**
Details to perform this change has been mentioned in the FALCON-1980.

### CODE REVIEW

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java [hunk 1]
```diff
@@ -163,13 +164,16 @@ private void addOutputFeedsAsArgument(List<String> argList, Cluster cluster) thr
             return;
         }
 
-        for(Output output : entity.getOutputs().getOutputs()) {
+        int numOutputFeed = entity.getOutputs().getOutputs().size();
```
Comment (pavankumar526) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: I can understand you want to put input first and then output , but with in output or input ordering should be maintained right ?

Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: @pavankumar526  Being I want input first and then output, I am doing reverse insertion. Inserting output first at 0 index and then input at 0 index in the arg List. So with this, final List will appear in order as input, output and user's argument. As number of arguments are not much, so there will not be much impact in List insertion at 0 index.

**Comment by pavankumar526** [comment]
+1 looks good

**Comment by peeyushb** [comment]
Thanks @pavankumar526 for the review. If no further comments, I will commit this patch soon.

File: oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java (hunk 1 above)
Comment (vrangan) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: Can you add a comment on this (that we are adding to the 0th index and getting the args shifted as arguments are added to get the desired effect).

Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java]: Comment added.

**Pull request closed without merging**